File: felt/node/utils/contracts.py
"""Module for loading contract ABI and address from build files."""
import json
from pathlib import Path

BUILD_FOLDER = Path(__file__).parent.parent / "build"

chain_id_map = {1337: "dev"}


def load_contracts(w3):
    """Load all deployed contracts on current chain from build folder."""
    chain_id = w3.eth.chain_id
    chain_id = chain_id_map.get(chain_id, chain_id)
    deploy_map = json.load((BUILD_FOLDER / "deployments/map.json").open())

    contracts = {}
    for key, address in deploy_map[chain_id].items():
        contract = json.load((BUILD_FOLDER / f"contracts/{key}.json").open())
        contracts[key] = w3.eth.contract(address=address[-1], abi=contract["abi"])

    return contracts

File: webapp/src/utils/contracts.js
import { Contract, ContractFactory } from 'ethers';

import map from '../artifacts/deployments/map.json';

function getChain(chainId) {
  return chainId === 1337 ? 'dev' : chainId;
}

export function getContractAddress(chainId, name) {
  const chain = getChain(chainId);
  try {
    return map[chain][name].at(-1);
  } catch (e) {
    console.log(
      `Couldn't find any deployed contract "${name}" on the chain "${chain}".`,
    );
    return undefined;
  }
}

export async function loadContract(chainId, name, library) {
  const chain = getChain(chainId);
  const address = getContractAddress(chainId, name);

  if (address === undefined) {
    return undefined;
  }

  let contractArtifact;
  try {
    contractArtifact = await import(
      `../artifacts/deployments/${chain}/${address}.json`
    );
  } catch (e) {
    console.log(
      `Failed to load contract artifact "../artifacts/deployments/${chain}/${address}.json"`,
    );
    return undefined;
  }

  return new Contract(address.substr(2), contractArtifact.abi, library);
}

export async function getContractFactory(name, signer) {
  let contractArtifact;
  try {
    contractArtifact = await import(`../artifacts/contracts/${name}.json`);
  } catch (e) {
    console.log(
      `Failed to load contract artifact "../artifacts/contracts/${name}.json"`,
    );
    return undefined;
  }

  return new ContractFactory(
    contractArtifact.abi,
    contractArtifact.bytecode,
    signer,
  );
}

export async function getProjectContract(address, library) {
  let contractArtifact;
  try {
    contractArtifact = await import(
      `../artifacts/contracts/ProjectContract.json`
    );
  } catch (e) {
    console.log(
      `Failed to load contract artifact "../artifacts/contracts/ProjectContract.json"`,
    );
    return undefined;
  }

  return new Contract(address, contractArtifact.abi, library);
}
